instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/test/dynamodbattributes_test.go

```diff
@@ -0,0 +1,409 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/aws/smithy-go/middleware"
+)
+
+func TestDynamodbTags(t *testing.T) {
+	cases := struct {
+		responseStatus     int
+		expectedRegion     string
+		expectedStatusCode int
+		expectedError      codes.Code
+	}{
+		responseStatus:     200,
+		expectedRegion:     "us-west-2",
+		expectedStatusCode: 200,
+	}
+
+		server := httptest.NewServer(http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(cases.responseStatus)
+			}))
+		defer server.Close()
+
+		t.Run("dynamodb tags", func(t *testing.T) {
+			sr := tracetest.NewSpanRecorder()
+			provider := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(sr))
+
+			svc := dynamodb.NewFromConfig(aws.Config{
+				Region: cases.expectedRegion,
+				EndpointResolverWithOptions: aws.EndpointResolverWithOptionsFunc(
+					func(service, region string, _ ...interface{}) (aws.Endpoint, error) {
+						return aws.Endpoint{
+							URL:         server.URL,
+							SigningName: "dynamodb",
+						}, nil
+					},
+				),
+				Retryer: func() aws.Retryer {
+					return aws.NopRetryer{}
+				},
+			})
+			_, err := svc.GetItem(context.Background(), &dynamodb.GetItemInput{
+				TableName:            aws.String("table1"),
+				ConsistentRead:       aws.Bool(false),
+				ProjectionExpression: aws.String("test"),
+				Key: map[string]dtypes.AttributeValue{
+					"id": &dtypes.AttributeValueMemberS{Value: "test"},
+				},
+			}, func(options *dynamodb.Options) {
+				otelaws.AppendMiddlewares(
+					&options.APIOptions, otelaws.WithAttributeSetter(otelaws.DynamodbAttributeSetter), otelaws.WithTracerProvider(provider))
+			})
+
+			if cases.expectedError == codes.Unset {
+				assert.NoError(t, err)
+			} else {
+				assert.NotNil(t, err)
+			}
+
+			spans := sr.Ended()
+			require.Len(t, spans, 1)
+			span := spans[0]
+
+			assert.Equal(t, "DynamoDB", span.Name())
+			assert.Equal(t, trace.SpanKindClient, span.SpanKind())
+			attrs := span.Attributes()
+			assert.Contains(t, attrs, attribute.Int("http.status_code", cases.expectedStatusCode))
+			assert.Contains(t, attrs, attribute.String("aws.service", "DynamoDB"))
+			assert.Contains(t, attrs, attribute.String("aws.region", cases.expectedRegion))
+			assert.Contains(t, attrs, attribute.String("aws.operation", "GetItem")) 
+			assert.Contains(t, attrs, attribute.String("aws.dynamodb.table_names", "table1"))
+			assert.Contains(t, attrs, attribute.String("aws.dynamodb.projection", "test"))
+			assert.Contains(t, attrs, attribute.Bool("aws.dynamodb.consistent_read", false))
+
+		})
+
+
+}
+
+func TestDynamodbTagsBatchGetItemInput(t *testing.T) {
```

Comment:
It looks like the functions from here down are pure unit tests of the attribute setter.  These should be moved up one level to live alongside the implementation.  The only tests that need to live in this module are integration tests that use the SDK, such as the one above.